UnityProject/Assets/Scripts/Items/SuitSensor.cs

```diff
@@ -144,34 +153,46 @@ public string GetInfo()
 			return sensorReport.ToString();
 		}
 
-
-		public IDCard GetIdentification()
+		private IDCard GetIdentification()
 		{
-			var SlotsOccupied = itemStorage.GetOccupiedSlots();
-			foreach (var SlotOccupied in SlotsOccupied)
+			var slotsOccupied = itemStorage.GetOccupiedSlots();
+			foreach (var slotOccupied in slotsOccupied)
 			{
-				var IDCard = SlotOccupied.Item.GetComponentCustom<IDCard>();
-				if (IDCard == null)
+				var idCard = slotOccupied.Item.GetComponentCustom<IDCard>();
+				if (idCard == null)
 				{
-					var PDA = SlotOccupied.Item.GetComponentCustom<PDALogic>();
+					var PDA = slotOccupied.Item.GetComponentCustom<PDALogic>();
 					if (PDA != null)
 					{
-						IDCard = PDA.GetIDCard();
+						idCard = PDA.GetIDCard();
 					}
 				}
-
-				if (IDCard != null)
+				if (idCard != null)
 				{
-					return IDCard;
+					return idCard;
 				}
 			}
 
+			if (pickupable.ItemSlot.RootPlayer() != null)
+			{
+				foreach (var slotId in pickupable.ItemSlot.RootPlayer().PlayerScript.Equipment.ItemStorage.GetNamedItemSlots(NamedSlot.id))
+				{
+					if (slotId.IsEmpty) continue;
+					return slotId.Item.GetComponent<IDCard>();
+				}
+			}
 			return null;
-
 		}
 
-		private void SwitchMode()
+		[Command(requiresAuthority = false)]
+		private void SwitchMode(PlayerScript player)
 		{
+			if (Vector3.Distance(gameObject.AssumedWorldPosServer(), player.AssumedWorldPos) > 3.5)
+			{
+				Loggy.LogWarning("[MedicalTerminal/SwitchMode] - Prevented possible cheating from player who is far away from this option.");
```

Comment:
Log the player?

Reply:
Good point, will add it when im back home

Reply:
Done